pkg/tfbridge/info.go

```diff
@@ -590,10 +598,16 @@ func (m *MarshallableProviderInfo) Unmarshal() *ProviderInfo {
 		dataSources[k] = v.Unmarshal()
 	}
 
-	return &ProviderInfo{
+	info := ProviderInfo{
 		P:           m.Provider.Unmarshal(),
 		Config:      config,
 		Resources:   resources,
 		DataSources: dataSources,
 	}
+
+	if m.TFProviderVersion != "" {
```

Comment:
Any reason to do this conditionally? just setting the field in the initializer will have the same effect. Same feedback above.

Reply:
You are correct - just being over cautious here - moved it back to the initializer